game/core/game_view.py

```diff
@@ -214,14 +240,17 @@ def reload_modules(self) -> None:
     @beartype
     def on_update(self, delta_time: float) -> None:
         """Incremental redraw."""
-        if self.pressed_keys.on_update():
-            self.on_key_hold()
-        self.rpg_movement.on_update()
-        self.game_map.on_update()
-        game_clock = self.game_clock.on_update(delta_time)
-        for register in self.get_all_registers():
-            if register.on_update:
-                register.on_update(game_clock)
+        try:
+            if self.pressed_keys.on_update():
+                self.on_key_hold()
+            self.rpg_movement.on_update()
+            self.game_map.on_update()
+            game_clock = self.game_clock.on_update(delta_time)
+            for register in self.get_all_registers():
+                if register.on_update:
+                    register.on_update(game_clock)
+        except Exception as exc:  # pylint: disable=broad-except
+            self.gui.draw_message_box(message=str(exc))
```

Comment:
We could add a log here:

```suggestion
	    logger.exception("Found error in on_update")
            self.gui.draw_message_box(message=str(exc))
```

Reply:
The purpose of this is to bubble up exceptions from the tasks (although it would also catch other exceptions as well). I intentionally don't want to mention the function because the player isn't supposed to look at the source code for the tasks except task 3.